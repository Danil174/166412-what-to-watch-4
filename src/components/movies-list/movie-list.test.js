import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list.jsx";

const testFilmsList = [
  {
    id: 11,
    title: `Orlando`,
    src: `img/orlando.jpg`,
    source: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 12,
    title: `Pulp Fiction`,
    src: `img/pulp-fiction.jpg`,
    source: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 13,
    title: `Revenant`,
    src: `img/revenant.jpg`,
    source: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 14,
    title: `Seven Years in Tibet`,
    src: `img/seven-years-in-tibet.jpg`,
    source: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

describe(`MoviesListTest`, () => {
  it(`render MoviesList`, () => {
    const tree = renderer
      .create(
          <MoviesList
            filmsList = {testFilmsList}
            onCardMouseEnter={() => {}}
            onCardMouseLeave={() => {}}
            onTitleOrImgClickHandler={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
