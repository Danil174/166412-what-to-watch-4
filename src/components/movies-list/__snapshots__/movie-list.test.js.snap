// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesListTest render MoviesList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Aviator"
        height="175"
        src="img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian Rhapsody"
        height="175"
        src="img/bohemian-rhapsody.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Dardjeeling Limited"
        height="175"
        src="img/dardjeeling-limited.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Dardjeeling Limited
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts The Crimes Of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts The Crimes Of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Johnny English"
        height="175"
        src="img/johnny-english.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Johnny English
      </a>
    </h3>
  </article>
</div>
`;
